lib/core/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -2387,3 +2440,388 @@ outputsCreatedByTx tx
         F.toList (collateralOutput tx)
     | otherwise =
         outputs tx
+
+--------------------------------------------------------------------------------
+-- Generating and shrinking arbitrary sequences of blocks
+--------------------------------------------------------------------------------
+
+-- | A sequence of blocks that starts at a given block height and slot number.
+--
+-- Values of this type model a sequence of state transitions affecting the
+-- global UTxO set:
+--
+-- @
+--    global_utxo_0 -> block_0_1 ->
+--    global_utxo_1 -> block_1_2 ->
+--    ...
+--    global_utxo_i -> block_i_j ->
+--    global_utxo_j
+-- @
+--
+-- In turn, each block contains a series of transactions that each affect the
+-- global UTxO set. The effect of applying a single block to the global UTxO
+-- set is equivalent to applying all of the contained transactions in order
+-- to the global UTxO set.
+--
+-- In general, only a subset of transactions within a block sequence will be
+-- relevant to a particular wallet, and only a subset of entries within each
+-- successive iteration of the global UTxO set will be relevant to that wallet.
+--
+-- Values of 'BlockSeq' are more convenient to generate and shrink than values
+-- of 'BlockData', but can easily be transformed into values of 'BlockData' in
+-- order to test properties of the 'applyBlocks' function.
+--
+-- Usage:
+--
+--  1. Use function 'genBlockSeq' to generate a 'BlockSeq' value.
+--  2. Use function 'blockSeqToBlockData' to convert a 'BlockSeq' value to a
+--     value of 'BlockData' suitable for use with the 'applyBlocks' function.
+--  3. Use function 'shrinkBlockSeq' to shrink to a minimal counterexample.
+--
+data BlockSeq = BlockSeq
+    { initialBlockHeight
+        :: Quantity "block" Word32
+    , initialSlotNo
+        :: SlotNo
+    , shrinkableTxSeq
+        :: ShrinkableTxSeq
```

Comment:
```suggestion
    { initialBlockHeight :: Quantity "block" Word32
    , initialSlotNo :: SlotNo
    , shrinkableTxSeq :: ShrinkableTxSeq
```

Reply:
@Unisay I think your suggested style is also fine, though the existing style doesn't violate the Coding Standards (to my knowledge). This style is also used in quite a few of our existing modules. Perhaps this discussion could be held outside the context of this PR?